blocks/init/src/blocks/custom/card-list/card-list.php

```diff
@@ -16,14 +16,9 @@
 $media_position = $attributes['mediaPosition'] ?? '';
 $button         = $attributes['button'] ?? [];
 
-$component_class  = "
-  {$block_class}
-  {$block_class}__media-position--{$media_position}
-";
-
 ?>
 
-<div class="<?php echo esc_attr( $component_class ); ?>">
+<div class="<?php echo esc_attr( "{$block_class} {$block_class}__media-position--{$media_position}" ); ?>">
```

Comment:
If we try to make .php and React block / component templates similar we should follow the same guiding principle / rules.

This should not be done inline but rather a few lines above (similar to how it was before), using `Components::classnames()` preferably (like you'd do in React).

Reply:
I would like to remove all those separate variables for classes before the html, they are hard to read and you need to jump up and down to change something. I'm ok with assigning this to a variable if it is something complicated but a s simple as this is I will like to have it in a inline form

Reply:
Fair enough but note we just did the exact oposite change in React components in this PR (moved inline code above `render` method) 😄

Reply:
can you please give me an example? 
are you referring to the check for this:
```const buttonObject = (typeof button === 'undefined') || button;```

Reply:
> `const buttonObject = (typeof button === 'undefined') || button;` 
yes, that was inline before (inside markup / render), we moved it above `render`.

In for example `card-list.php`, classes were defined above markup. Now they're moved inline. 

It's the exact oposite of the `const buttonObject = (typeof button === 'undefined') || button;` type changes.

Reply:
I'm generally fine when it's simple (like here, only 2 classes) but we're just adding more inconsistencies between the way we write React and the way we write PHP components.

Reply:
Yes but this is different. The changes for the undefined was done because you don't want a conditions inside a props that you are passing inside a component. As for this example this is only css selectors inline there are no conditions here. I there were some conditions I would move it out of the render methods just like in the PHP.

you can see here that Kuki also used classes inside the html inline:
https://infinum.slack.com/archives/C027S03F6/p1588695511204800

Reply:
> I'm generally fine when it's simple (like here when it's only 2 classes) but we're just adding more inconsistencies between the way we write React and the way we write PHP components.

Can we agree that if the selectors became less readable inline you can extract them inside a variable before. If there are few like here where is 2 you can leave it inline?

Reply:
> Can we agree that if the selectors became less readable inline you can extract them inside a variable before. If there are few like here where is 2 you can leave it inline?

Agreed.